### PR TITLE
Assorted stuff

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by martinpitt** [comment]
The autokpgtest logs are inaccessible, I'll investigate/fix this. -- Indeed it seems there are problems with Openstack swift in our cloud, our admins are working on this.

**Comment by martinpitt** [comment]
Swift is back, thus the logs work again. The failure is

```
======================================================================
FAIL: test_tmp_mount (__main__.ServicesTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/autopkgtest.TTV0xS/build.1pH/systemd/debian/tests/boot-and-services", line 130, in test_tmp_mount
    self.assertEqual(status, 3, status_out)
AssertionError: 4 != 3 : Failed to dump process list, ignoring: Unit tmp.mount not found.
● tmp.mount
   Loaded: not-found (Reason: No such file or directory)
   Active: inactive (dead)
```

from this test: http://anonscm.debian.org/cgit/pkg-systemd/systemd.git/tree/debian/tests/boot-and-services#n111 .

This was introduced in PR #3385 (which got merged despite a test regression), so it doesn't affect _this_ PR.

**Comment by martinpitt** [comment]
@poettering : What do you think about adding a label "ignore-ci-failure" for such cases where we reviewed CI regressions and justified them to be okay/good enough to land the PR?

File: src/libsystemd-network/sd-dhcp-server.c [hunk 1]
```diff
@@ -633,7 +633,8 @@ static int parse_request(uint8_t code, uint8_t len, const void *option, void *us
 
                 break;
         case SD_DHCP_OPTION_MAXIMUM_MESSAGE_SIZE:
-                if (len == 2)
+
+                if (len == 2 && unaligned_read_be16(option) >= sizeof(DHCPPacket))
                         req->max_optlen = unaligned_read_be16(option) - sizeof(DHCPPacket);
```
Comment (keszybz) [inline src/libsystemd-network/sd-dhcp-server.c:639]: I wonder if we shouldn't reject the packet instead. But maybe that's too harsh. Simply ignoring the invalid option sounds OK too.

File: shell-completion/bash/journalctl [hunk 2]
```diff
@@ -79,7 +68,7 @@ _journalctl() {
                                 comps='short short-iso short-precise short-monotonic verbose export json json-pretty json-sse cat'
                         ;;
                         --field|-F)
-                                comps=${__journal_fields[*]}
+                                comps=$(journalctl --fields | sort 2>/dev/null)
```
Comment (keszybz) [inline shell-completion/bash/journalctl:71]: Yeah, good catch. This code predated `--fields`, and should have been updated when `--fields` was added.

**Pull request merged**